dotcom-rendering/src/web/components/PinnedPostLiveness.importable.tsx

```diff
@@ -25,9 +27,41 @@ export const PinnedPostLiveness = ({}) => {
 		pinnedPost && pinnedPost.scrollHeight <= pinnedPost.clientHeight;
 	if (contentFitsContainer) hideShowMore();
 
+	if (pinnedPost) {
```

Comment:
I wanted to emit the INSERT event only when pinned post is loaded in the page. Meaning we only want to emit 1 event at the beginning. I'm not sure if it's right to just emit the event like this, and I think maybe, we need to create a isLoaded state or something like that to make sure we only emit the insert event when the page loads.

Right now there's no state in this component PinnedPostLiveness so from what I see, there should not be any re-rendering.

What do you think is the best way to tackle this?

Reply:
I'm not sure if I'm reading this right but how does this sound?

If this event represents the pinned post becoming visible (appears in viewport), then I guess you could say it's the same as duration? You will never have a duration value without the post becoming visible and every post that is visible also has a duration. So... do we even need this event?

If D&I need to have a sort of count unique for page views that had a pinned post become visible, then I think this is possible as a query on the data. This would be preferable to the browser having to send more data.

Reply:
Great points. Most probably as you said we may not even need to send this data any more. Let me share this with the team and also with D & I. Will update the decision here later

Reply:
Just before writing this to the D & I, wanted to double check something with you. You mentioned that through query on the data we can get the count unique for page views that had a pinned post? 

But if a page with pinned post never emits the `duration` event meaning the pinned post was never viewed, how does the D & I find out if that page had a pinned post (if we were to stop emitting `INSERT` events)?

Reply:
🤔 I thought the INSERT event was only fired if the post came into view? In which case the duration would always also be sent

Reply:
Of course, if it is the opposite and we want to track every page view that included a pinned post, then can we derive this information from the plain page view data? Or maybe using a simple `data-component-name` attribute? I'm not super sure how the `tracker-js` script works but maybe there's an easy solution there?

Reply:
Thanks @oliverlloyd for pointing this out. After some digging in with @alinaboghiu and chatting with @rtyley I think the solution could be as simple as adding `data-component="pinned-post"` to the `PinnedPost` component. I just ran and tested it using the ophan tester tool and the following is showing up in the page view data:
![image](https://user-images.githubusercontent.com/15894063/158374277-a007c80b-5d3a-4615-b9d3-821f414ce360.png)

Reply:
@oliverlloyd what do you think about this change? If you are happy with it, I also prepared the same change in frontend for the pinned post https://github.com/guardian/frontend/pull/24763

Reply:
Amazing! This looks a lot simpler 👍